assets/src/edit-story/components/form/numeric.js

```diff
@@ -97,48 +111,124 @@ function Numeric({
   canBeEmpty,
   ...rest
 }) {
-  const isMultiple = value === MULTIPLE_VALUE;
-  const placeholder = isMultiple ? __('multiple', 'web-stories') : '';
-  const [dot, setDot] = useState(false);
   const [inputValue, setInputValue] = useState(value);
-  const ref = useRef();
 
-  useEffect(() => setInputValue(value), [value]);
+  const inputRef = useRef();
+  const targetFormRef = useRef(null);
+  const skipValidationRef = useRef(false);
+
+  const { focused, handleFocus, handleBlur } = useFocusAndSelect(inputRef);
+
+  const isMultiple = inputValue === MULTIPLE_VALUE;
+  const placeholder = isMultiple ? __('multiple', 'web-stories') : '';
+
+  const handleChange = useCallback((event) => {
+    setInputValue(event.target.value);
+  }, []);
+
+  const validateAndSubmitInput = useCallback(() => {
+    targetFormRef.current = inputRef.current.form;
+
+    const validValue = validateInput(inputValue, {
+      canBeNegative,
+      canBeEmpty,
+      float,
+    });
+
+    setInputValue(validValue ?? value);
+    onChange(validValue ?? value);
+  }, [inputValue, canBeNegative, canBeEmpty, float, value, onChange]);
 
   const handleUpDown = useCallback(
     ({ key, altKey }) => {
       if (isMultiple) {
         return;
       }
 
-      let newValue;
+      const validValue = validateInput(inputValue, {
+        canBeNegative,
+        canBeEmpty,
+        float,
+      });
+
+      // The `|| 0` is to prevent newValue from being set to
+      // empty string by either validValue or value
+      let newValue = (validValue ?? value) || 0;
       const diff = Big(float && altKey ? 0.1 : 1);
       if (key === 'ArrowUp') {
         // Increment value
         newValue =
           typeof max !== 'undefined'
-            ? Math.min(max, Big(value).plus(diff))
-            : Big(value).plus(diff);
+            ? Math.min(max, Big(newValue).plus(diff))
+            : Big(newValue).plus(diff);
       } else if (key === 'ArrowDown') {
         // Decrement value
         newValue =
           typeof min !== 'undefined'
-            ? Math.max(min, Big(value).minus(diff))
-            : Big(value).minus(diff);
+            ? Math.max(min, Big(newValue).minus(diff))
+            : Big(newValue).minus(diff);
+      }
+
+      setInputValue(newValue.toString());
+    },
+    [isMultiple, float, max, inputValue, value, min, canBeNegative, canBeEmpty]
+  );
+
+  const handleKeyDownEffect = useCallback(
+    ({ key, altKey }) => {
+      switch (key) {
+        case 'Escape':
+          // Revert input value and exit input focus without
+          // triggering blur validation
+          setInputValue(value);
+          skipValidationRef.current = true;
+          inputRef.current.blur();
```

Comment:
And here as well, I think we should select the entire input field contents after reverting.

Reply:
Okay.  But there are existing Karma tests that specifically test that hitting escape on an input field should lose focus.  Currently, if you have an item selected in the editor and then put your focus on a numeric input field.  If you hit escape, the field would lose focus, and if you hit escape _again_ the selected element would be deselected.  

But with this change, things should work like this when a field is focused:
- Hit `esc` and value reverts, all input in field is selected.
- Hit `esc` again, and if all text is selected, then input loses focus.

Right?  Double `esc` to lose focus on input so we can continue the pattern of `esc` backing you out of your selections.

Reply:
I can completely live with `esc` blurring the input field. That's actually fine with me - seems consistent with other editors.